samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -355,6 +355,7 @@ class JobModelManager(
    */
   val localityManager: LocalityManager = null) extends Logging {
 
+  val localityMappings : Map[String, String] = scala.collection.mutable.Map[String, String]()
```

Comment:
Does this need to be mutable? Can we read the locality mappings up front and then pass them directly to the JobModelManager?